### PR TITLE
tech: Add stack layout patterns

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Added new utility functions (`stack`, `hStack`, `vStack`) to the theme patterns for managing flexbox layouts with ease.

### What changed?

- Created a new file `patterns.ts` in the `src/theme` directory.
- Added `stack`, `hStack`, `vStack` functions for flexbox layout configurations.
- Updated `sprinkles.css.ts` to include `gap` property in responsive properties.

## Justify why they are needed
To simplify the implementation of flexbox layouts in UI components by providing reusable utility functions.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
* **#4352** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4352?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/theme/patterns.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  direction?: 'row' | 'column'
+} & Omit<Sprinkles, 'display' | 'flexDirection'>
+
+export const stack = ({ direction = 'column', gap = 'md', ...params }: StackPatternParams) => {
+  return sprinkles({ display: 'flex', flexDirection: direction, gap, ...params })
+}
+
+type DirectedStackPatternParams = Omit<StackPatternParams, 'direction'>
+
+export const hStack = (params: DirectedStackPatternParams) => {
```
Comment (gustaveen) [inline packages/ui/src/theme/patterns.ts:14]: What do you think about `xStack` and `yStack`? I'm kind of used to the pattern we had for the `Space` component :)

**Review by gustaveen — COMMENTED** [review]
Nice! 🙌 

Was just thinking about the naming

File: packages/ui/src/theme/patterns.ts (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/theme/patterns.ts:14]: Personally I am used to the H and V naming more than X and Y but both work for me. I'll update.

**Review by gustaveen — APPROVED** [review]
🥳

Regarding naming, I'm good with what you think is best, what's your preference @guilhermespopolin @alebedev @falklinus

**Review by guilhermespopolin — APPROVED** [review]
Super nice :)
I think the names are good 🚀

File: packages/ui/src/theme/patterns.ts (hunk 1 above)
Comment (alebedev) [reply packages/ui/src/theme/patterns.ts]: `column` and `row` ?

Comment (Youakeem) [reply packages/ui/src/theme/patterns.ts]: It's now merged with x and y. Feel free to update it if you want, I don't mind any of them.

**Pull request merged**